File: python/mykoob/models/lesson.py
class Lesson:
    def __init__(self, data: dict):
        self.number = data.get("lesson_number", None)
        self.timetable_id = data.get("timetableid", None)
        self.time_from = data.get("time_from", None)
        self.time_to = data.get("time_to", None)
        self.discipline = data.get("discipline", None)
        self.teacher = data.get("teacher", None)
        self.notes = data.get("notes", None)
        self.theme = self.theme = None if data.get("theme", None) == "" else data.get("theme", None)
        self.last_changes: str = data.get("last_changes", None)
        self.classroom = data.get("classroom", None)

    def __repr__(self):
        return (f"Lesson(number={self.number}, discipline={self.discipline}, teacher={self.teacher}, classroom={self.classroom}), theme={self.theme}, last_changes={self.last_changes})")

File: javascript/src/models/lesson.js
class Lesson {
    constructor(data) {
        this.number = data.lesson_number || null;
        this.timetableId = data.timetableid || null;
        this.timeFrom = data.time_from || null;
        this.timeTo = data.time_to || null;
        this.discipline = data.discipline || null;
        this.teacher = data.teacher || null;
        this.notes = data.notes || null;
        this.theme = data.theme === "" ? null : data.theme || null;
        this.lastChanges = data.last_changes || null;
        this.classroom = data.classroom || null;
    }

    toString() {
        return `Lesson(number=${this.number}, discipline=${this.discipline}, teacher=${this.teacher}, classroom=${this.classroom}, theme=${this.theme}, last_changes=${this.lastChanges})`;
    }
}

export { Lesson };
